automation/check-patch.e2e-linux-bridge-functests.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+set -xeu
+
+# This script should be able to execute linux-bridge
+# functional tests against Kubernetes cluster with
+# CNAO built with latest changes, on any
+# environment with basic dependencies listed in
+# check-patch.packages installed and docker running.
+#
+# yum -y install automation/check-patch.packages
+# automation/check-patch.e2e-linux-bridge-functests.sh
+
+teardown() {
+    cp $(find . -name "*junit*.xml") $ARTIFACTS || true
+    cluster/kubectl.sh logs -n cluster-network-addons -l app=cni-linux-bridge-plugin > $ARTIFACTS/linux-bridge.log || true
+    cluster/kubectl.sh get nodes -o yaml > $ARTIFACTS/nodes.yaml || true
+    rm -rf "${TMP_COMPONENT_PATH}"
+    cd ${TMP_PROJECT_PATH}
+    make cluster-down
+}
+
+build_repo_binaries() {
+	./build_linux.sh
+}
+
+collect_binaries_shasum_from_repo() {
+	sha_file=$1
+  sha256sum bin/* >${sha_file}
+}
+
+collect_binaries_shasum_from_node() {
+	sha_file=$1
+	${TMP_PROJECT_PATH}/cluster/ssh.sh node01 -- sha256sum /opt/cni/bin/* >${sha_file}
+}
+
+extract_sha_from_shasum_file() {
+	file_name_in_sha=$1
+	sha_file=$2
+	grep ${file_name_in_sha1} ${sha_file1} | awk '{print $1}'
+}
+
+assert_sha_files_eq() {
+	sha_file1=$1
+	file_name_in_sha1=$2
+	sha_file2=$3
+	file_name_in_sha2=$4
+
+	file1_sha=$(extract_sha_from_shasum_file ${file_name_in_sha1} ${sha_file1})
+	file2_sha=$(extract_sha_from_shasum_file ${file_name_in_sha2} ${sha_file2})
+
+	if [[ ${file1_sha} !=  ${file2_sha} ]]; then
+		echo "files ${file_name_in_sha1},${file_name_in_sha2} do not match shasum"
+		exit 1
+	else
+		echo "files ${file_name_in_sha1},${file_name_in_sha2} have the same shasum"
+	fi
+}
+
+main() {
+    # Setup CNAO and artifacts temp directory
+    source automation/check-patch.setup.sh
+    cd ${TMP_PROJECT_PATH}
+
+    # Spin-up ephemeral cluster with latest CNAO
+    # this script also exports KUBECONFIG, and fetch $COMPONENT repository
+    COMPONENT="linux-bridge" source automation/components-functests.setup.sh
+
+    trap teardown EXIT
+
+    echo "Run linux-bridge functional tests"
+    (
```

Comment:
This `()` execs another sub-shell. Why this is needed?

Reply:
not really needed. removed